writer/licence_writer.go

```diff
@@ -54,3 +58,68 @@ func CheckIfLicenceFormatIsValid(file *os.File) bool {
 	}
 	return true
 }
+
+func WriteToFileLicence(fileName string) (bool, error) {
+	ok, err := CheckIfFileContainsLicenceAlready(fileName)
+
+	if ok {
+		return false, err
+	}
+
+	// make a temporary outfile
+	outfile, err := os.Create("temp.java")
```

Comment:
Why are you creating a temporary file here. Is it necessary ?

Reply:
Yes it is. What we are doing here is that we firstly write licence string to the top of the temporary file then reading original file and writing that file also. Because this approach is much easier than replacing existing file.